lsp-mode.el

```diff
@@ -5506,27 +5506,37 @@ perform the request synchronously."
                         symbol))
                     document-symbols)))
     (if children?
-        (cons symbol (lsp--document-symbols->symbols-hierarchy children? current-position))
+        (cons symbol (lsp--document-symbols->document-symbols-hierarchy children? current-position))
       (when symbol
         (list symbol)))))
 
-(defun lsp--symbols-informations->symbols-hierarchy (symbols-informations current-position)
+(lsp-defun lsp--symbol-information->document-symbol ((&SymbolInformation :name :kind :location :container-name? :deprecated?))
+  "Convert a SymbolInformation to a DocumentInformation"
+  (lsp-make-document-symbol :name name
+                            :kind kind
+                            :range (lsp:location-range location)
+                            :children? nil
+                            :deprecated? deprecated?
+                            :selection-range (lsp:location-range location)
+                            :detail? container-name?))
+
+(defun lsp--symbols-informations->document-symbols-hierarchy (symbols-informations current-position)
   "Convert SYMBOLS-INFORMATIONS to symbols hierarchy on CURRENT-POSITION."
   (seq-filter (-lambda ((symbol &as &SymbolInformation :location (&Location :range (&Range :start start-position
```

Comment:
This is insufficient since it doesn't actually perform the expected conversion. You need something like

``` elisp
(mapcan (-lambda ((symbol &as &SymbolInformation :location (&Location :range (&Range :start start-position
                                                                                       :end end-position))))
            (when (and (lsp--position-compare current-position start-position)
                       (lsp--position-compare end-position current-position))
              (list (lsp--symbol-information->document-symbol symbol))))
          symbols-informations)
```

or the equivalent `cl-loop` list comprehension.